ceremonies/src/tests.rs

```diff
@@ -2301,10 +2367,189 @@ fn unregistering_newbie_with_no_participants_works() {
+		EncointerCommunities::insert_bootstrappers(cid, vec![bootstrapper.clone()]);
+		assert!(EncointerBalances::issue(cid, &bootstrapper, NominalIncome::from_num(1)).is_ok());
+
+		let a = AccountKeyring::Alice.pair();
+		let alice = account_id(&a);
+
+		let proof = make_reputable_and_get_proof(&a, cid, cindex - 1);
+		assert_ok!(register(alice.clone(), cid, Some(proof)));
+		assert_eq!(EncointerCeremonies::reputable_registry((cid, cindex), 1).unwrap(), alice);
+		assert_eq!(EncointerCeremonies::reputable_count((cid, cindex)), 1);
+		assert_eq!(
+			EncointerCeremonies::participant_reputation((cid, cindex), &alice),
+			Reputation::UnverifiedReputable
+		);
+		assert_eq!(
+			EncointerCeremonies::participant_reputation((cid, cindex - 1), &alice),
+			Reputation::VerifiedLinked
+		);
+
+		assert_ok!(EncointerCeremonies::unregister_participant(
+			Origin::signed(alice.clone()),
+			cid,
+			Some((cid, cindex - 1))
+		));
+
+		assert!(!ParticipantReputation::<TestRuntime>::contains_key((cid, cindex), &alice));
+		assert_eq!(
+			EncointerCeremonies::participant_reputation((cid, cindex - 1), &alice),
+			Reputation::VerifiedUnlinked
+		);
+		assert_eq!(EncointerCeremonies::reputable_count((cid, cindex)), 0);
+	})
+}
+
+#[test]
+fn unregister_participant_fails_with_reputables_and_wrong_reputation() {
+	new_test_ext().execute_with(|| {
+		let cid = register_test_community::<TestRuntime>(None, 0.0, 0.0);
+		let cindex = EncointerScheduler::current_ceremony_index();
+		IssuedRewards::<TestRuntime>::insert((cid, cindex - 1), 0, ());
+		let bootstrapper = account_id(&AccountKeyring::Ferdie.pair());
+		EncointerCommunities::insert_bootstrappers(cid, vec![bootstrapper.clone()]);
+		assert!(EncointerBalances::issue(cid, &bootstrapper, NominalIncome::from_num(1)).is_ok());
+
+		let a = AccountKeyring::Alice.pair();
+		let alice = account_id(&a);
+
+		let proof = make_reputable_and_get_proof(&a, cid, cindex - 1);
+		assert_ok!(register(alice.clone(), cid, Some(proof)));
+		assert_eq!(EncointerCeremonies::reputable_registry((cid, cindex), 1).unwrap(), alice);
+		assert_eq!(EncointerCeremonies::reputable_count((cid, cindex)), 1);
+		assert_eq!(
+			EncointerCeremonies::participant_reputation((cid, cindex), &alice),
+			Reputation::UnverifiedReputable
+		);
+		assert_eq!(
+			EncointerCeremonies::participant_reputation((cid, cindex - 1), &alice),
+			Reputation::VerifiedLinked
+		);
+
+		assert_err!(
+			EncointerCeremonies::unregister_participant(Origin::signed(alice.clone()), cid, None),
+			Error::<TestRuntime>::ReputationCommunityCeremonyRequired,
+		);
+
+		EncointerCeremonies::fake_reputation(
+			(cid, cindex - 1),
+			&alice,
+			Reputation::VerifiedUnlinked,
+		);
+
+		assert_err!(
+			EncointerCeremonies::unregister_participant(
+				Origin::signed(alice.clone()),
+				cid,
+				Some((cid, cindex - 1))
+			),
+			Error::<TestRuntime>::ReputationMustBeLinked,
+		);
+	})
+}
+
+#[test]
+fn unregister_participant_works_with_newbies() {
```

Comment:
Same, we are missing `unregister_participant_fails_in_invalid_phase`, `unregister_participant_fails_with_inexistent_community`